control-plane/pkg/reconciler/base/reconciler.go

```diff
@@ -153,6 +153,10 @@ func (r *Reconciler) createDataPlaneConfigMap(ctx context.Context) (*corev1.Conf
 		},
 	}
 
+	if r.DataPlaneConfigMapTransformer != nil {
+		r.DataPlaneConfigMapTransformer(cm)
+	}
```

Comment:
Good catch!

We can probably just delete the same call in line 137 as that sets the labels on in-memory resources, but let's keep it for stability. Maybe there's some code which persists that in-memory resources.